### PR TITLE
osd: check OSDSuperblock in mkfs() when it already have superblock

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
ref: http://tracker.ceph.com/issues/13586

add test-case to reproduce this problems.
(do mkfs when an existing superblock)

the following is the fixed description:

    When we remove the ready flag from osd directory, it will execute mkfs again.
    It will read superblock but can not check osd id and cluster fsid correctly.

    It need to decode the superblock that we read from osd.
    It will always failure now, so I modify it for read osd id and fsid correctly.

Signed-off-by: Vicente Cheng \<freeze.bilsted@gmail.com\>

### CODE REVIEW

**Comment by dachary** [comment]
@Vicente-Cheng could you please add this test to https://github.com/ceph/ceph/pull/6367 instead of making a separate pull request ? Thanks !

**Comment by Vicente-Cheng** [comment]
@dachary 
this pull request including this test and fixed.
I still need to add this test to #6376 ? or just use this pull request?

**Comment by dachary** [comment]
no, I was just confused, all is good

**Comment by Vicente-Cheng** [comment]
@dachary 
I afraid build failed, so I pull the latest version to repush it.
sorry to make you confused :p

**Comment by dachary** [comment]
@Vicente-Cheng no worries, this is a great fix ;-)

File: src/test/osd/osd-reactivate.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# GNU Library Public License for more details.
+#
+
+source ../qa/workunits/ceph-helpers.sh
+
+function run() {
+    local dir=$1
+    shift
+
+    export CEPH_MON="127.0.0.1:7106"
```
Comment (dachary) [inline src/test/osd/osd-reactivate.sh]: Use 7122 instead of 7106 here so that it does not conflict with another port

Comment (Vicente-Cheng) [inline src/test/osd/osd-reactivate.sh]: OK!

I didn't notice that.
thanks for reminder!

Comment (dachary) [inline src/test/osd/osd-reactivate.sh]: @Vicente-Cheng you could not know. I added reminders at https://github.com/ceph/ceph/pull/6387, how does that look ?

Comment (Vicente-Cheng) [inline src/test/osd/osd-reactivate.sh]: @dachary 
that's a good reminder for someone want to write a new test function!!

btw your comment is `# git grep '\<7109\> : there must be only one`
maybe lost another `'`?
like that  `# git grep '\<7109\>' : there must be only one`

anyway, I think that reminder is useful!!

**Comment by dachary** [comment]
<pre>
Reviewed-by: Loic Dachary &lt;ldachary@redhat.com>
</pre>

after a rados suite passe.

**Comment by dachary** [comment]
@Vicente-Cheng are you willing to run a rados suite ?

**Comment by Vicente-Cheng** [comment]
@dachary 
Sure, but I don't know how to run a rados suite.
would you tell me some samples to let me know how to complete it?

thanks!!!
vicente

**Pull request merged**